src/components/search/Search.svelte

```diff
@@ -1,41 +1,64 @@
 <script>
-  import { tick } from 'svelte';
+import { tick } from 'svelte';
 import { fly } from 'svelte/transition';
+import { cubicOut } from 'svelte/easing';
 import SearchIcon from 'udgl/icons/Search.svelte';
 import LineSegSpinner from 'udgl/LineSegSpinner.svelte';
+import { throttle } from 'throttle-debounce';
 
-import {
-    store,
-} from '../../state/store';
-
-
-import telemetrySearch from '../../state/telemetry-search';
-
+import { getSearchResults } from '../../state/api';
 import TelemetrySearchResults from './SearchResults.svelte';
 
 
 let inputElement;
 let searchContainer;
+let results = [];
+let searchIsActive = false;
+let searchQuery = '';
+
+const SEARCH_THROTTLE_TIME = 500; // how often to send the PSS fetch (in ms)
+const SEARCH_RESULTS_LIMIT = 15; // maximum number of results to show
 
 function turnOnSearch() {
-    store.setField('searchIsActive', true);
+  searchIsActive = true;
 }
 
 function unfocus() {
-    inputElement.blur();
-    store.setField('searchIsActive', false);
+  inputElement.blur();
+  searchIsActive = false;
 }
 
 async function onKeypress(event) {
```

Comment:
Small thing, but since we only need the key from the event, we could save one line by doing:

```javascript
async function onKeypress({ key }) {
```

Personal preference one way or the other. Not a big deal.